### PR TITLE
Improve support for self-hosted clusters

### PR DESCRIPTION
**Pull request by ThisIsQasim**

### CODE REVIEW

**Comment by jodevsa** [comment]
sorry for the delays; I'll merge this tomorrow morning

File: controllers/wireguard_controller.go [hunk 1]
```diff
@@ -739,6 +749,17 @@ func (r *WireguardReconciler) deploymentForWireguard(m *vpnv1alpha1.Wireguard, i
 									MountPath: "/tmp/wireguard/",
 								}},
 						}},
+					InitContainers: []corev1.Container{
+						{
+							SecurityContext: &corev1.SecurityContext{
+								Privileged:  &privileged,
```
Comment (jodevsa) [inline controllers/wireguard_controller.go:755]: lets make this optional as some devs don't have access to run privileged containers and ip_forward is mostly enabled everywhere

Comment (jodevsa) [reply controllers/wireguard_controller.go:755]: can be triggered using something like:
```
apiVersion: vpn.example.com/v1alpha1
kind: Wireguard
metadata:
  name: "my-cool-vpn"
spec:
  enable_ip_forwrd_on_init:  True
```

**Comment by jodevsa** [comment]
Thank you so much for contributing!

File: controllers/wireguard_controller.go (hunk 1 above)
Comment (jodevsa) [reply controllers/wireguard_controller.go:755]: I don't have strong opinions here; so let me know your thoughts :)

**Comment by jodevsa** [comment]
closed in favor of: https://github.com/jodevsa/wireguard-operator/pull/25 ; your commits will be merged with that PR :)

File: controllers/wireguard_controller.go (hunk 1 above)
Comment (ThisIsQasim) [reply controllers/wireguard_controller.go:755]: Yeah that would work.

**Comment by ThisIsQasim** [comment]
Thanks for taking this up!

**Pull request closed without merging**